indico/modules/events/registration/badges.py

```diff
@@ -209,3 +200,263 @@ def _build_pdf(self, canvas):
                 x_cm = (self.width - x*cm - self.tpl_data.width_cm*cm)
                 self._draw_badge(canvas, person, self.template.backside_template,
                                  self.backside_tpl_data, x_cm, y * cm)
+
+
+class GoogleWalletManager:
+    """Google Wallet management class.
+
+    """
+
+    def __init__(self, event, registration=None):
+        self.event = event
+        self.registration = registration
+        self.settings = self.get_google_wallet_settings(self.event.category)
+        self.credentials = None
+        self.http_client = None
+        self.base_url = 'https://walletobjects.googleapis.com/walletobjects/v1'
+        self.batch_url = 'https://walletobjects.googleapis.com/batch'
+        self.class_url = f'{self.base_url}/eventTicketClass'
+        self.object_url = f'{self.base_url}/eventTicketObject'
+
+        # Set up authenticated client
+        if self.settings:
+            self.settings['google_wallet_application_credentials'] = json.loads(
+                self.settings['google_wallet_application_credentials'])
+            self.auth()
+
+    @staticmethod
+    def get_google_wallet_settings(category):
+        while category:
+            if category.google_wallet_settings:
+                return category.google_wallet_settings
+            else:
+                category = category.parent
+        return {}
+
+    def auth(self):
+        """Create authenticated HTTP client using a service account file."""
+        try:
+            self.credentials = Credentials.from_service_account_info(
+                self.settings['google_wallet_application_credentials'],
+                scopes=['https://www.googleapis.com/auth/wallet_object.issuer'])
+            self.http_client = AuthorizedSession(self.credentials)
+        except Exception as e:
+            Logger.get('events.registration').warning('Cannot Authenticate to Google Wallet link: %s', str(e))
```

Comment:
And probably any code that relies on the credential will just fail. So I'd rather not catch this exception and rely on the standard exception handling logic.